packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -691,5 +690,5 @@ Future<DateTime> showDatePicker({
       lastDate: lastDate
     )
   );
-  return picked ?? initialDate;
+  return picked ?? null;
```

Comment:
There's no need to `?? null`.  You can just return `picked`.  The `??` operator returns the right-hand side when the left-hand side is null.  If the left-hand side is null, then we can just return that null.